EIPS/EIPs/eip-5075.md

```diff
@@ -0,0 +1,294 @@
+---
+eip: 5075
+title: "RateLimit: an outflow limiter for assets"
+description: Limits outflows for all contract assets to a given rate in a given timeframe
+author: PR0 (@pr0toshi)
+discussions-to: https://ethereum-magicians.org/t/eip-x-limit-token-outflows-within-timeframe-to-limit-hack-losses/9172
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022--05-05
+---
+
+## Abstract
+An inheritable implementation for a rate limiter using a new transferL fn which replaces all asset outflow transfer fns to unknown external addresses. By limiting asset outflows for all assets in a contract within a customisable timeframe, contracts which suffer from hacks will have losses limited to the given rateLimit param, which is dynamic and relative to a contracts net total assets, being tracked at the per asset level.
```

Comment:
```suggestion
An inheritable implementation for a rate limiter using a new `transferL` function which replaces all asset outflow transfer functions to unknown external addresses. By limiting asset outflows for all assets in a contract within a customizable timeframe, contracts that suffer from hacks will have losses limited to the given `rateLimit` parameter, which is dynamic and relative to a contract's net total assets, being tracked at the per asset level.
```

I personally would also prefer if the function name was more descriptive, say `transferWithLimit` or `transferLimited`.